src/functionaltest/java/com/ericsson/ei/config/SeleniumConfig.java

```diff
@@ -1,31 +1,65 @@
 import org.openqa.selenium.firefox.FirefoxDriver;
 import org.openqa.selenium.firefox.FirefoxOptions;
 import org.openqa.selenium.firefox.FirefoxProfile;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Properties;
 
 import java.io.File;
 
 public class SeleniumConfig {
-
     private static final Logger LOGGER = LoggerFactory.getLogger(SeleniumConfig.class);
 
+    private static String propertiesPath = String.join(File.separator, "src", "functionaltest", "resources");
+    private static String propertiesFile = "functional-test.properties";
+
+    private static String firefoxBZip2FileUrlLinux = "";
+    private static String firefoxBZip2FileNameLinux = "";
+
     private static File tempDownloadDirectory = Files.createTempDir();
+    private static FirefoxDriver driver;
 
-    public static FirefoxDriver getFirefoxDriver() throws OSNotSupportedException {
-        FirefoxDriver driver;
-        FirefoxProfile firefoxProfile = new FirefoxProfile();
+    public static FirefoxDriver initFirefoxDriver() throws PropertiesNotLoadedException, OSNotSupportedException {
+        if (driver != null) {
+            return driver;
+        }
 
+        FirefoxProfile firefoxProfile = new FirefoxProfile();
         firefoxProfile.setPreference("browser.download.folderList", 2);
         firefoxProfile.setPreference("browser.download.dir", tempDownloadDirectory.getPath());
         firefoxProfile.setPreference("browser.helperApps.neverAsk.saveToDisk", "application/json");
 
+        FirefoxOptions firefoxOptions = new FirefoxOptions()
+                .setHeadless(true)
+                .setProfile(firefoxProfile);
+
+        boolean successfullyLoadedProperties = loadProperties();
+        if (!successfullyLoadedProperties) {
+            LOGGER.error("Properties was not properly loaded.");
+            throw new PropertiesNotLoadedException();
+        };
+
         if (SystemUtils.IS_OS_LINUX) {
+            String firefoxBZip2FilePath = String.join(
+                    File.separator, tempDownloadDirectory.getPath(), firefoxBZip2FileNameLinux);
+            Utils.downloadFileFromUrlToDestination(firefoxBZip2FileUrlLinux, firefoxBZip2FilePath);
+            Utils.extractBZip2InDir(firefoxBZip2FilePath, tempDownloadDirectory.getPath());
+            File firefoxBinaryFilePath = new File(
+                    String.join(File.separator, tempDownloadDirectory.getPath(), "firefox", "firefox"));
+            Utils.makeBinFileExecutable(firefoxBinaryFilePath);
+            FirefoxBinary firefoxBinary = new FirefoxBinary(firefoxBinaryFilePath);
+            firefoxOptions.setBinary(firefoxBinary);
+
             System.setProperty("webdriver.gecko.driver", "src/functionaltest/resources/geckodriver");
         } else if (SystemUtils.IS_OS_WINDOWS) {
             System.setProperty("webdriver.gecko.driver", "src/functionaltest/resources/geckodriver.exe");
```

Comment:
Small detail but this OS not supported logging could be LOGGER.error(SystemUtils.OS_NAME + " currently not supported.") There is also an OS_VERSION that could be used.

Reply:
Done